### PR TITLE
feat: configurable Contract naming

### PR DESCRIPTION
**Pull request by anupamvs**
**Issue-** Currently Contract Naming is done Party Name. Since, autoname() function is overridden in contract controller.
**Changes-**
- Added Naming Series field.
- Introduced Config in CRM Settings-
Screenshot 2022-03-06 at 4 22 33 PM
docs- https://docs.erpnext.com/docs/v14/user/manual/en/CRM/crm_settings#2-contract-naming-by

### CODE REVIEW

File: erpnext/crm/doctype/contract/contract.py [hunk 1]
```diff
@@ -5,22 +5,27 @@
...
+		if frappe.db.get_single_value("CRM Settings", "contract_naming_by") == "Naming Series":
+			set_name_by_naming_series(self)
 
-		if self.contract_template:
-			name += " - {} Agreement".format(self.contract_template)
+		else:
+			name = self.party_name
+
+			if self.contract_template:
+				name += " - {} Agreement".format(self.contract_template)
```
Comment (rohitwaghchaure) [inline erpnext/crm/doctype/contract/contract.py]: use f string

File: erpnext/crm/doctype/contract/contract.py [hunk 2]
```diff
@@ -5,22 +5,27 @@
...
 
-		# If identical, append contract name with the next number in the iteration
-		if frappe.db.exists("Contract", name):
-			count = len(frappe.get_all("Contract", filters={"name": ["like", "%{}%".format(name)]}))
-			name = "{} - {}".format(name, count)
+			# If identical, append contract name with the next number in the iteration
+			if frappe.db.exists("Contract", name):
+				count = len(frappe.get_all("Contract", filters={"name": ["like", "%{}%".format(name)]}))
+				name = "{} - {}".format(name, count)
```
Comment (rohitwaghchaure) [inline erpnext/crm/doctype/contract/contract.py]: use f string

File: erpnext/crm/doctype/contract/contract.py [hunk 3]
```diff
@@ -5,22 +5,27 @@
...
+			if self.contract_template:
+				name += " - {} Agreement".format(self.contract_template)
 
-		# If identical, append contract name with the next number in the iteration
-		if frappe.db.exists("Contract", name):
-			count = len(frappe.get_all("Contract", filters={"name": ["like", "%{}%".format(name)]}))
-			name = "{} - {}".format(name, count)
+			# If identical, append contract name with the next number in the iteration
+			if frappe.db.exists("Contract", name):
+				count = len(frappe.get_all("Contract", filters={"name": ["like", "%{}%".format(name)]}))
```
Comment (rohitwaghchaure) [inline erpnext/crm/doctype/contract/contract.py]: use frappe.qb to get count

**Comment by ankush** [comment]
@rohitwaghchaure this is already merged in v13: https://github.com/frappe/erpnext/pull/30450

**Pull request closed without merging**